drivers/timer/arcv2_timer0.c

```diff
@@ -50,7 +50,8 @@
 #define MAX_TICKS ((COUNTER_MAX / CYC_PER_TICK) - 1)
 #define MAX_CYCLES (MAX_TICKS * CYC_PER_TICK)
 
-#define TICKLESS (IS_ENABLED(CONFIG_TICKLESS_KERNEL))
+#define TICKLESS (IS_ENABLED(CONFIG_TICKLESS_KERNEL) &&		\
+		  !IS_ENABLED(CONFIG_QEMU_TICKLESS_WORKAROUND))
```

Comment:
this was removed due to icount introduction.

Reply:
@nashif sure I will remove entire commit `drivers: arcv2_timer0: Add work-around for QEMU` (https://github.com/zephyrproject-rtos/zephyr/pull/25100/commits/dd6b3a69919f3dc5661b9b74025bd42c0ebb1c69) then.

But the problem is we don't yet support `icount` (it's todo list but no ETAs yet for it) :(

Reply:
Be advised that lack of icount support does cause problems. We have quite a few timing sensitive tests that sporadically fall over if the host system is heavily loaded and QEMU keeps trying to sync the emulated clock to host time

Reply:
> @nashif sure I will remove entire commit `drivers: arcv2_timer0: Add work-around for QEMU` ([dd6b3a6](https://github.com/zephyrproject-rtos/zephyr/commit/dd6b3a69919f3dc5661b9b74025bd42c0ebb1c69)) then.
> 
> But the problem is we don't yet support `icount` (it's todo list but no ETAs yet for it) :(

if you need that workaround to pass the tests given you do not have icount I would suggest to reintroduce the workaround for ARC while we wait for icount.

Reply:
@nashif for starters I guess for ARC QEMU we may just disable `TICKLESS` config option until `icount` is implemented. What do you think?

Reply:
@nashif with rebase on top of the current master, removal of ARC Timer fix-up for QEMU & exclusion of `qemu_arc{em|hs}` in some tests we now have clean results, see https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/73352/